tools/metadatastore-upgrade/README.md

```diff
@@ -0,0 +1,35 @@
+# MetadataStore Upgrade Tool
+
```

Comment:
how about add a sentence that user has these options:
(1) upgrade whole KFP system (via CLI guide/doc), so naturally MLMD server will be upgraded (MLMD server handles MLMD DB's migration).
(2) if user don't want to upgrade whole KFP system, then follow this tool to just upgrade MLMD server (MLMD server handles MLMD DB's migration as long as --enable_database_upgrade=true which is suggested).

I may didn't get the reason on why we provide this tool. I thought "kubectl apply" with MLMD new manifest would be easier and more acceptable by k8s admin.

Is it more for the reason that we didn't set "--enable_database_upgrade=true" before and want to help existing user correct it? If so, I think a simple yaml just for MLMD with "kubectl apply" can also help on it.

Reply:
"--enable_database_upgrade=true" 
Yuan mentioned you two synced on this part that if enable this flag for MLMD gRPC server, the MLMD scheme will be dirty? If so, I think it's an issue MLMD team should fix, right?

@dushyanthsc @Bobgy

Reply:
@rmgogogo The "--enable_database_upgrade=true" is not supported to be set by default. The MLMD has a work item to fix this. In the meantime we provide this tool to the OSS community to be able to run pipelines that have been compiled (outside the cluster) with different version of MLMD.

Reply:
I have also updated the README with your suggestion.

Reply:
got it, thanks.

https://github.com/google/ml-metadata/blob/master/ml_metadata/metadata_store/metadata_store_server_main.cc#L173

For new installations. If we enable this flag in our YAML, for the new installations, it will be enabled and then user no need to manually use this tool, right? Aka, as long as user install a newer version, all done automatically.

https://github.com/kubeflow/pipelines/search?q=enable_database_upgrade&unscoped_q=enable_database_upgrade